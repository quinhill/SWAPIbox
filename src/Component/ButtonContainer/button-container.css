.button {
  height: 30px;
  width: 100px;
  color: black;
  background-color: #FFC343;
  border: none;
  margin: 20px 50px 20px 50px;
}