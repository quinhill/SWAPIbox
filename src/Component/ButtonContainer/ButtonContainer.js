import React from 'react';
import './button-container.css';
import PropTypes from 'prop-types';

const ButtonContainer = ({checkState}) => {

  const handleClick = (e) => {
    e.preventDefault()
    const name = e.target.name
    checkState(name)
  }

  return (
    <div className="button-container">
      <button 
        className="button" 
        id="people" 
        name="people" 
        onClick={handleClick}
      >
        people
      </button>
      <button 
        className="button" 
        id="planets"
        name="planets"
        onClick={handleClick}
      >
        planets
      </button>
      <button 
        className="button" 
        id="vehicles"
        name="vehicles"
        onClick={handleClick} 
      >
        vehicles
      </button>
    </div>
  )
}

ButtonContainer.Proptypes = {
  getData: PropTypes.func.isRequired
}

export default ButtonContainer;