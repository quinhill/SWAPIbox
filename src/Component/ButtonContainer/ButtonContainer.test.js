import React from 'react';
import ButtonContainer from './ButtonContainer';
import { shallow } from 'enzyme';

describe('ButtonContainer', () => {
  
  let mockEvent;
  let mockGetData;
  let wrapper;

  beforeEach(() => {
    mockEvent = { preventDefault: jest.fn(), target: { name: 'people' } }
    mockGetData = jest.fn()

    wrapper = shallow(<ButtonContainer
                        getData={mockGetData}
                        />)
  })

  it('should call getData with the correct params', () => {
    const expected = 'people'

    wrapper.instance().handleClick()

    expect(mockGetData).toHaveBeenCalledWith(expected)
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})