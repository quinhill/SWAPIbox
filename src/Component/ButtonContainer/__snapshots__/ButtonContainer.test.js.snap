// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonContainer should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonContainer
    getData={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="button"
          id="people"
          name="people"
          onClick={[Function]}
>
          people
</button>,
        <button
          className="button"
          id="planets"
          name="planets"
          onClick={[Function]}
>
          planets
</button>,
        <button
          className="button"
          id="vehicles"
          name="vehicles"
          onClick={[Function]}
>
          vehicles
</button>,
      ],
      "className": "button-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "people",
          "className": "button",
          "id": "people",
          "name": "people",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "people",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "planets",
          "className": "button",
          "id": "planets",
          "name": "planets",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "planets",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "vehicles",
          "className": "button",
          "id": "vehicles",
          "name": "vehicles",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "vehicles",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="button"
            id="people"
            name="people"
            onClick={[Function]}
>
            people
</button>,
          <button
            className="button"
            id="planets"
            name="planets"
            onClick={[Function]}
>
            planets
</button>,
          <button
            className="button"
            id="vehicles"
            name="vehicles"
            onClick={[Function]}
>
            vehicles
</button>,
        ],
        "className": "button-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "people",
            "className": "button",
            "id": "people",
            "name": "people",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "people",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "planets",
            "className": "button",
            "id": "planets",
            "name": "planets",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "planets",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "vehicles",
            "className": "button",
            "id": "vehicles",
            "name": "vehicles",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "vehicles",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
